### PR TITLE
Updated codebase to ES2015

### PR DESCRIPTION
**Pull request by jwngr**
**PLEASE REVIEW BUT DO NO MERGE INTO `master`.**

This is the first prerequisite step for ReactFire `2.0.0`. This updates the entire codebase to use ES2015 and get transpiled via Babel. I intentionally chose to not fix the lint issues in the source and test files since it would be a lot of wasted work since hardly any of that code is going to be re-used in ReactFire `2.0.0.`

I also updated almost every dependency to the latest and simplified the ESLint configuration to use the default used by the Airbnb style guide.

### CODE REVIEW

File: gulpfile.babel.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  ],
+};
+
+
+/*
+ * TASKS
+ */
+// Lints the JavaScript files
+gulp.task('lint', () => {
+  // Temporarily skip linting of source and test files on teh ReactFire 2.0 branch
```
Comment (tstirrat) [inline gulpfile.babel.js]: typo

File: gulpfile.babel.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    .pipe(eslint.failAfterError());
+});
+
+
+/* Builds the distribution files */
+gulp.task('build', () => {
+  gulp.src(paths.srcFiles)
+    // Convert to ES5
+    .pipe(babel({
+      presets: ['react', 'es2015'],
```
Comment (tstirrat) [inline gulpfile.babel.js]: is this needed if it's also in the .babelrc?

Comment (jwngr) [inline gulpfile.babel.js]: Good catch. Doesn't look like I need this here.

**Comment by jwngr** [comment]
Thanks for the review! I fixed both things, but I'm assigning this back to myself until I get Travis passing.

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,11 @@
 language: node_js
 node_js:
-- '0.10'
-- '0.12'
+- '4'
+- '6'
 # - stable
 sudo: false
+env:
```
Comment (jwngr) [inline .travis.yml]: This junk (and the `addons` below) is needed to get certain Babel modules to properly compile on Node.js 4+.

**Comment by jwngr** [comment]
Woot, Travis is passing! Back to you @tstirrat!

File: .travis.yml (hunk 3 above)
Comment (tstirrat) [inline .travis.yml]: I'd put a comment inline so that one day the junk might be removed.

**Comment by jwngr** [comment]
Back to you again @tstirrat.

**Comment by tstirrat** [comment]
LGTM! 

🎾

**Comment by jwngr** [comment]
Thanks! I'm going to keep this branch around and use it to track the ReactFire `2.0.0` effort. More PRs hopefully coming sooner rather than later.

**Comment by theobouwman** [comment]
Why not merge this right now?

**Comment by jwngr** [comment]
There's nothing to merge yet. All this does is update the code to use ES2015. It doesn't affect the output library or add any new features. It is simply a setup step in order to get a `2.0.0` release out.

**Pull request merged**